### DISCUSSION TITLE
Trouble extending configuration from local package in monorepo

### DISCUSSION THREAD

**Discussion by bacaxnot** [eslint/eslint#18101] [Q&A] [answered]
**Environment**
- ESLint Version: `8.56.0`
- Node.js Version: `18.17.0`
- npm/pnpm Version: `8.10.5`
- Operating System: `MacOS Sonoma 14.2.1`

**Affected Plugins**
- `@typescript-eslint/parser`: `6.17.0`
- `@typescript-eslint/eslint-plugin`: `6.17.0`
- Other plugins/configurations included in `@bacaxnot/config`: `@vercel/style-guide`, `eslint-config-turbo`, `eslint-config-prettier`, `eslint-plugin-only-warn`

**ESLint Configuration**
- Location: `/apps/docs/.eslintrc.js`
```javascript
/** @type {import("eslint").Linter.Config} */
module.exports = {
  root: true,
  extends: ["@bacaxnot/config/eslint/next.js"],
  parser: "@typescript-eslint/parser",
  parserOptions: {
    project: true,
  },
};
```

- `/packages/config/eslint/next.js` configuration:
```javascript
const { resolve } = require("node:path");
const project = resolve(process.cwd(), "tsconfig.json");

/** @type {import("eslint").Linter.Config} */
module.exports = {
  extends: [
    "eslint:recommended",
    "prettier",
    require.resolve("@vercel/style-guide/eslint/next"),
    "eslint-config-turbo",
  ],
  globals: {
    React: true,
    JSX: true,
  },
  env: {
    node: true,
    browser: true,
  },
  plugins: ["only-warn"],
  settings: {
    "import/resolver": {
      typescript: {
        project,
      },
    },
  },
  ignorePatterns: ["*.js", "node_modules/"],
  overrides: [{ files: ["*.js?(x)", "*.ts?(x)"] }],
};
```

**Issue Description**
When running `pnpm lint` in my monorepo setup, ESLint is unable to find the configuration file `@bacaxnot/config/eslint/next.js` that I am trying to extend in my `.eslintrc.js`. The error message indicates that the config cannot be found, even though the path and exports seem correct.

**Steps to Reproduce**
1. Clone the monorepo from [GitHub Repo Link](https://github.com/bacaxnot/monorepo).
2. Run `pnpm installl` at the root folder.
3. Navigate to the `docs` project within the monorepo.
4. Run `pnpm lint` to execute ESLint.

**What I've Tried**
- Ensured that `@bacaxnot/config` is correctly symlinked in `docs/node_modules`.
- Verified the `files` field in `@bacaxnot/config/package.json` to include the `eslint` directory.
- Used `require.resolve()` in the ESLint config to ensure paths are correctly resolved.
- Cleared `pnpm` cache and reinstalled dependencies.
- Checked ESLint, Node.js, and pnpm versions for compatibility issues.

**Request for Help**
I'm looking for guidance on why ESLint is unable to resolve the configuration from the local package within our monorepo and any steps I might have missed in ensuring the correct resolution of the ESLint config file.

**Thank You**
I appreciate the time and effort of the volunteers helping with this issue. Please let me know if any additional information is required.

**Comment by Rec0iL99** [comment, marked as answer]
Hi @bacaxnot, this is happening because the ESLint shareable config is not following the naming convention enforced by ESLint for [shareable configs](https://eslint.org/docs/latest/extend/shareable-configs#creating-a-shareable-config). Your config name should be `@bacaxnot/eslint-config`.

**Reply by bacaxnot** [reply]
I see... I've spent so many hours looking for the answer... Thanks!!
